src/main/java/net/md_5/specialsource/Jar.java

```diff
@@ -75,8 +75,17 @@
      * @return
      */
     public boolean containsClass(String clazz) {
+        if (contains.contains(clazz)) {
+            return true;
+        }
+
         try {
-            return contains.contains(clazz) ? true : getClass(clazz) != null;
+            InputStream current = getClass(clazz);
```

Comment:
If you're modifying this, then you may as well not be using the input stream and check add proper contains methods for classes / resources to eliminate the waste of opening an inputstream and closing it on the next line.

Reply:
😆 
We nearly simultaneously commented about the same thing.

I'll change it tonight.